### PR TITLE
Build bounty catalog

### PR DESCRIPTION
**Pull request by asherbig**
Alright this took some time. When I say that the todo-list module is independent from the rest of the app, I mean it in almost every way possible. I think the only things I re-use are the auth-service, and the destiny cache service. I probably even wrap some of the functionality in my own services. I know it's not IDEAL, but I feel like the organization really helps me work faster since I'm familiar with it all, and I get to use those _sweet sweet_ rxjs observable operators.

No UI elements added yet. This PR is working out the backing data. The focus here is the bounties. The bounty-catalog service builds a list of unique bounties (for a whole account). The `chars` property on the `Bounty` object stores any character-specific bounty info. I'm _extremely_ happy with the result and feel like I don't have much bloat in these files right now.

### CODE REVIEW

File: src/app/todo-list/components/todo-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  ) {
+    super()
+  }
+
+  ngOnInit(): void {
+    this.bountyService.bountyCatalog.pipe(
+      filter(x => !!x),
+      takeUntil(this.destroy$)
+    ).subscribe((bounties) => {
+      console.log('Account bounties with possesion/completion statuses', bounties);
```
Comment (asherbig) [inline src/app/todo-list/components/todo-list.component.ts:26]: You can see the list of bounties in the console once everything completes. The network requests are still the biggest time-suck, even after requesting ONLY the components I want. Will remove it when I add UI elements.

File: src/app/todo-list/interfaces/vendor.interface.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  progress?: any; // TODO: track progress and define a better interface
+}
+
+export enum SaleStatus {
+  AVAILABLE = 0,
+  COMPLETED = 8,
+  NOT_FOR_SALE = 32,
+  NOT_AVAILABLE = 64,
+  ALREADY_HELD = 128,
+}
```
Comment (asherbig) [inline src/app/todo-list/interfaces/vendor.interface.ts:102]: Most of the things in this file are exposing properties already on the objects we get back from the API or the manifest.

File: src/app/todo-list/services/bounty-catalog.service.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  private fetchBountiesForChars() {
+    this.context.characters.pipe(
+      filter((x) => !!x),
+      switchMap((chars) => {
+        this.chars = chars;
+        const dataStream = this.createVendorsStream(chars);
+        return combineLatest(...dataStream);
+      }),
```
Comment (asherbig) [inline src/app/todo-list/services/bounty-catalog.service.ts:51]: Once the context service has the characters loaded, we can go fetch the vendor info. There's a TON of info that I don't really need, but I don't think I can request less data and still get the saleStatus for all bounties for all characters. Either way, this way is code-easy, but not as performant as possible, which I'm definitely fine with for MVP

File: src/app/todo-list/services/context-service.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  constructor(
+    private http: HttpService,
+    private bungieService: BungieService
+  ) {
+    super();
+    // does a one-time load of the auth key into the service.
+    // it's not very resilient to changes in the auth key, but it works
+    // for the basic case where it's there and it doesn't change
+    // it would be great if the auth service had a behaviorSubject that always stores
+    // the up-to-date auth key
```
Comment (asherbig) [inline src/app/todo-list/services/context-service.ts:36]: outdated comments that I'll clean up in a later PR

**Comment by dcaslin** [comment]
Looks good!

**Pull request merged**